torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -348,3 +360,121 @@ def _recurse_update_dict(module, destination, prefix):
     for submodule_name, submodule in module.named_modules():
         if submodule_name != '':
             _recurse_update_dict(submodule, destination, prefix + submodule_name + '.')
+
+def shard_parameter(
+        module: torch.nn.Module,
+        param_name: str,
+        sharding_spec: ShardingSpec,
+        src_rank=0,
+        process_group=None):
+    """
+    Given a :class:`torch.nn.Module`, a ``param_name`` for a parameter in that
+    module, it shards that parameter according to the provided
+    ``sharding_spec``. ``src_rank`` denotes the source rank which would be
+    used as the ground truth of the data which would be scattered as shards
+    across the rest of the ranks.
+
+    This method replaces ``module.param_name`` with a
+    :class:`torch.distributed._sharded_tensor.ShardedTensor`
+
+    Args:
+        module (:class:`torch.nn.Module`): Module whose parameter needs to be sharded.
+        param_name (str): Name of the parameter of ``module`` that needs to be sharded.
+        sharding_spec (:class:`torch.distributed._sharding_spec.ShardingSpec`): The specification
+            describing how to shard the Tensor.
+        size (int...): a sequence of integers defining the shape of the output
+            tensor. Can be a variable number of arguments or a collection like a list or tuple.
+
+    Keyword args:
+        src_rank (int, optional): The source rank which is used as the ground truth of
```

Comment:
will this API only be used by sharding an already initialized Module? `src_rank` won't be useful when the Module not been initialized?